src/ast/passes/semantic_analyser.cpp

```diff
@@ -1977,6 +2002,13 @@ void SemanticAnalyser::visit(FieldAccess &acc)
   // A field access must have a field XOR index
   assert((acc.field.size() > 0) != (acc.index >= 0));
 
+  if (func_ == "offsetof")
+  {
+    LOG(ERROR, acc.loc, err_)
+		<< "offsetof not support subfield yet";
+    return;
+  }
```

Comment:
This can be covered when handling the call. Also it's not entirely correct as it'd prevent using field access in the first argument.

Reply:
> This can be covered when handling the call. Also it's not entirely correct as it'd prevent using field access in the first argument.

because `offsetof(record, x)` is different with all other `call`, `builtin`(special vargs), Some places require special handling, I'll try a generic way to see if it works

Reply:
If the `dynamic_cast` approach that I suggested above works, then you can throw an error when the second arg is not an identifier in the call visitor.